tests/core/test_variable.py

```diff
@@ -0,0 +1,558 @@
+        #test variable dependent on DesignVariable
+        v = relentless.DesignVariable(value=1.0)
+        w = relentless.SameAs(v)
+        self.assertEqual(w.value, 1.0)
+        self.assertCountEqual(w.params, ('a',))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'a':v})
+
+        #test variable dependent on DependentVariable
+        u = relentless.SameAs(w)
+        self.assertEqual(u.value, 1.0)
+        self.assertCountEqual(u.params, ('a',))
+        self.assertDictEqual({p:v for p,v in u.depends}, {'a':w})
+
+        #test invalid variable dependence
+        with self.assertRaises(TypeError):
+            u = relentless.SameAs(1.0)
+
+    def test_value(self):
+        """Test setting values."""
+        #create "chained" dependent variables
+        v = relentless.DesignVariable(value=1.0, low=0.5, high=1.5)
+        w = relentless.SameAs(v)
+        x = relentless.SameAs(w)
+
+        self.assertEqual(v.value, 1.0)
+        self.assertEqual(w.value, 1.0)
+        self.assertEqual(x.value, 1.0)
+
+        #change value of v
+        v.value = 0.4
+        self.assertEqual(v.value, 0.5)
+        self.assertEqual(w.value, 0.5)
+        self.assertEqual(x.value, 0.5)
+
+    def test_derivative(self):
+        """Test _derivative method."""
+        v = relentless.DesignVariable(value=1.0)
+        w = relentless.SameAs(v)
+
+        #test w.r.t. a
+        dw = w._derivative('a')
+        self.assertEqual(dw, 1.0)
+
+        #test w.r.t. ~a
+        with self.assertRaises(ValueError):
+            w._derivative('x')
+
+class test_ArithmeticMean(unittest.TestCase):
+    """Unit tests for relentless.ArithmeticMean"""
+
+    def test_init(self):
+        """Test creation with data."""
+        #test variable dependent on 2 DesignVariables
+        u = relentless.DesignVariable(value=1.0)
+        v = relentless.DesignVariable(value=2.0)
+        w = relentless.ArithmeticMean(u, v)
+        self.assertAlmostEqual(w.value, 1.5)
+        self.assertCountEqual(w.params, ('a','b'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'a':u,'b':v})
+
+        #test variable dependent on 1 DesignVar, 1 DependentVar
+        x = relentless.ArithmeticMean(v,w)
+        self.assertAlmostEqual(x.value, 1.75)
+        self.assertCountEqual(x.params, ('a','b'))
+        self.assertDictEqual({p:v for p,v in x.depends}, {'a':v,'b':w})
+
+        #test variable dependent on 2 DependentVariables
+        y = relentless.ArithmeticMean(w,x)
+        self.assertAlmostEqual(y.value, 1.625)
+        self.assertCountEqual(y.params, ('a','b'))
+        self.assertDictEqual({p:v for p,v in y.depends}, {'a':w,'b':x})
+
+        #test same-valued dependencies
+        z = relentless.DesignVariable(value=1.0)
+        w = relentless.ArithmeticMean(u, z)
+        self.assertAlmostEqual(w.value, 1.0)
+        self.assertCountEqual(w.params, ('a','b'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'a':u,'b':z})
```

Comment:
You can probably remove this test, it doesn't really check much different than above.